code/campaignclassic/src/main/java/com/adobe/marketing/mobile/BasicTemplateNotificationBuilder.java

```diff
@@ -125,6 +210,259 @@ class BasicTemplateNotificationBuilder {
+        final String titleTextColor =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.TITLE_TEXT_COLOR);
+        final String expandedBodyTextColor =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.EXPANDED_BODY_TEXT_COLOR);
+        final String smallIcon =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.SMALL_ICON);
+        final String smallIconColor =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.SMALL_ICON_COLOR);
+        final String customSound =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.CUSTOM_SOUND);
+        final String actionButtonsString =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.ACTION_BUTTONS_STRING);
+
+        final Notification notification =
+                createNotificationBuilder(
+                                context,
+                                expandedLayout,
+                                smallLayout,
+                                remindLaterTimestamp,
+                                badgeCount,
+                                visibility,
+                                importance,
+                                imageUri,
+                                actionUri,
+                                channelId,
+                                customSound,
+                                titleText,
+                                bodyText,
+                                expandedBodyText,
+                                notificationBackgroundColor,
+                                titleTextColor,
+                                expandedBodyTextColor,
+                                messageId,
+                                deliveryId,
+                                smallIcon,
+                                smallIconColor,
+                                actionButtonsString,
+                                remindLaterText)
+                        .build();
+
+        notificationManager.notify(messageId.hashCode(), notification);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    static void handleRemindIntent(final Context context, final Intent intent) {
+        // get basic notification values from the intent extras
+        final Bundle intentExtras = intent.getExtras();
+        if (intentExtras == null) {
+            Log.trace(
+                    CampaignPushConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Intent extras are null, will not schedule a notification from the received"
+                            + " intent with action %s",
+                    intent.getAction());
+            return;
+        }
+
+        // set the calender time to the remind timestamp to allow the notification to be displayed
+        // at the later time
+        final int remindLaterTimestamp =
+                (int) (intentExtras.getLong(CampaignPushConstants.IntentKeys.REMIND_TS));
```

Comment:
Any specific reason you are coercing this to an int instead of leaving it as a `long`?

Reply:
the method to add time to a calendar object takes an `int` only
```
calendar.add(Calendar.SECOND, secondsUntilFireDate);
```

Reply:
Can we do the calculation entirely in `long` and then cast the result it to `int` when invoking the method?
This approach will yield a more accurate result because the division is done with long precision before casting to int. It's generally recommended to perform calculations with the highest precision available (e.g., long) before casting to a lower precision type (e.g., int) if necessary.

Reply:
sounds good, will update